src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -48,125 +42,159 @@
+    }));
   }
 
   public void replay() {
-      clickElement(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
-      muted = false;
-      playing = true;
+    wait.forElementVisible(replayOverlaySelector).click();
   }
 
   public void close() {
-    findCloseButton().click();
+    wait.forElementClickable(closeButtonSelector).click();
+  }
+
+  public void closeWithJS() {
+    new JavascriptActions(driver).execute(
+        "arguments[0].click();",
+        driver.findElement(closeButtonSelector)
+    );
   }
 
   public void clickOnClickArea2() {
-    clickOnAdClickArea(AD_TNG_CLICK_AREA_2_SELECTOR);
+    clickElementInsideAd(By.cssSelector(AD_TNG_CLICK_AREA_2_SELECTOR));
   }
 
   public void clickOnClickArea4() {
-    clickOnAdClickArea(AD_TNG_CLICK_AREA_4_SELECTOR);
+    clickElementInsideAd(By.cssSelector(AD_TNG_CLICK_AREA_4_SELECTOR));
   }
 
   public void clickOnAdImageResolvedState() {
-    clickOnAdClickArea(AD_RESOLVED_STATE_IMAGE_SELECTOR);
+    clickElementInsideAd(By.cssSelector(AD_RESOLVED_STATE_IMAGE_SELECTOR));
   }
 
-  private void clickOnAdClickArea(String clickAreaSelector) {
-    usingAdFrame(() -> wait.forElementClickable(By.cssSelector(clickAreaSelector)).click());
-  }
-
-  public double getCurrentTime() {
-    final String currentTime = usingVideoContext(video -> video.getAttribute("currentTime"));
-    return Double.parseDouble(currentTime);
+  private void clickElementInsideAd(By selector) {
+    usingAdFrame(() -> {
+      JavascriptActions jsActions = new JavascriptActions(driver);
+      // It need to be clicked by JS, because our templates elements covers each other
+      // and there is no way to click it by just .click()
+      jsActions.execute("arguments[0].click();", driver.findElement(selector));
+    });
   }
 
   public double getVideoHeightWhilePaused() {
-    if (!playing) {
-      return driver.findElement(By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot))).getSize().getHeight();
-    }
-    return 0;
+    return driver.findElement(pauseOverlaySelector).getSize().getHeight();
   }
 
   public double getAdSlotHeight() {
     waitForAdToLoad();
-    return driver.findElement(By.cssSelector("#" + slot)).getSize().getHeight();
+    return driver.findElement(By.id(slot)).getSize().getHeight();
   }
 
   public int getProgressBarWidth() {
-    final String selector = String.format(CURRENT_TIME_SELECTOR_FORMAT, slot);
-    return driver.findElement(By.cssSelector(selector)).getSize().getWidth();
+    return driver.findElement(progressBarSelector).getSize().getWidth();
   }
 
   public boolean hasStarted() {
     return getCurrentTime() > 0;
   }
 
   public boolean isVisible() {
-    return usingVideoIframeContext(webDriver -> webDriver.findElement(By.tagName("video")).isDisplayed());
+    return usingImaBridge(webDriver -> webDriver.findElement(By.tagName("video")).isDisplayed());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "video" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)